File: CsharpProjects/TemplateMethodSolucao/TemplateMethodSolucao/MetodoPagamento/PagamentoDebito.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TemplateMethodSolucao.GatewayCobranca;

namespace TemplateMethodSolucao.MetodoPagamento
{
    public class PagamentoDebito : Pagamento
    {
        public PagamentoDebito(double valor, Gateway gateway)
        {
            base.valor = valor;
            base.gateway = gateway;
        }

        public override double calcularTaxa()
        {
            return 4;
        }
        public override double calcularDesconto()
        {
            return this.valor * 0.05;
        }
    }
}

File: JavaProjects/TemplateMethodSolucao/src/metodoPagamento/PagamentoDebito.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package metodoPagamento;

import gatewayPagamento.Gateway;

/**
 *
 * @author jeffe
 */
public class PagamentoDebito extends Pagamento {

    public PagamentoDebito(double valor, Gateway gateway) {
        super.valor = valor;
        super.gateway = gateway;
    }
    
    @Override
    public double calcularTaxa(){
        return 4;
    }
    
    @Override
    public double calcularDesconto(){
            return super.valor*0.05;
    }
    
}
